### PR TITLE
DateTimePicker: Enable default-time attribute in date picker

### PR DESCRIPTION
**Pull request by RickMacTurk**
PR for issue: https://github.com/ElemeFE/element/issues/10078

### CODE REVIEW

File: examples/docs/en-US/datetime-picker.md [hunk 1]
```diff
@@ -117,6 +118,15 @@ DateTimePicker is derived from DatePicker and TimePicker. For a more detailed ex
       :picker-options="pickerOptions1">
     </el-date-picker>
   </div>
+  <div class="block">
+    <span class="demonstration">With default time</span>
+    <el-date-picker
+      v-model="value3"
+      type="datetime"
+      placeholder="Select date and time"
+      :default-time="['12:00:00']">
```
Comment (wacky6) [inline examples/docs/en-US/datetime-picker.md]: Why not a value but an array?

File: packages/date-picker/src/panel/date.vue [hunk 2]
```diff
@@ -147,6 +147,7 @@
     isDate,
     modifyDate,
     modifyTime,
+    modifyWithGivenTime,
```
Comment (wacky6) [inline packages/date-picker/src/panel/date.vue]: perhaps `modifyWithTimeString` is a better name?

File: packages/date-picker/src/panel/date.vue [hunk 3]
```diff
@@ -214,7 +215,10 @@
     methods: {
       proxyTimePickerDataProperties() {
         const format = timeFormat => {this.$refs.timepicker.format = timeFormat;};
-        const value = value => {this.$refs.timepicker.value = value;};
+        const value = value => {
+          const defaultTime = this.defaultTime || [];
```
Comment (wacky6) [inline packages/date-picker/src/panel/date.vue]: As mentioned before, for non-range, defaultTime should be a single value, not an array

**Comment by RickMacTurk** [comment]
@wacky6 Made changes based on feedback.

**Comment by Leopoldthecoder** [comment]
image

A test case has failed.

File: examples/docs/en-US/datetime-picker.md [hunk 4]
```diff
@@ -117,6 +118,15 @@ DateTimePicker is derived from DatePicker and TimePicker. For a more detailed ex
       :picker-options="pickerOptions1">
     </el-date-picker>
   </div>
+  <div class="block">
+    <span class="demonstration">With default time</span>
+    <el-date-picker
+      v-model="value3"
+      type="datetime"
+      placeholder="Select date and time"
+      :default-time="'12:00:00'">
```
Comment (wacky6) [inline examples/docs/en-US/datetime-picker.md]: How about `default-time="12:00:00"`? It is cleaner

File: packages/date-picker/src/util/index.js [hunk 5]
```diff
@@ -142,6 +142,14 @@ export const modifyTime = function(date, h, m, s) {
   return new Date(date.getFullYear(), date.getMonth(), date.getDate(), h, m, s, date.getMilliseconds());
 };
 
+export const modifyWithDefaultTime = (date, time) => {
```
Comment (wacky6) [inline packages/date-picker/src/util/index.js]: How about `modifyWithTimeString`? DefaultTime does not imply argument type.

File: packages/date-picker/src/panel/date.vue [hunk 6]
```diff
@@ -318,7 +321,7 @@
 
       handleDatePick(value) {
         if (this.selectionMode === 'day') {
-          this.date = modifyDate(this.date, value.getFullYear(), value.getMonth(), value.getDate());
+          this.date = modifyWithDefaultTime(value, this.defaultTime);
```
Comment (wacky6) [inline packages/date-picker/src/panel/date.vue]: This resets time to `defaultTime` whenever a date is picked. Current behavior sets time only if time is not set.

Comparison: 
* (current) https://jsfiddle.net/zbgmnqod/1/
* (this pr) https://jsfiddle.net/bmxe9u8r/

@Leopoldthecoder What's your thoughts on this behavior?

Comment (wacky6) [reply packages/date-picker/src/panel/date.vue]: Might need https://github.com/ElemeFE/element/issues/9878 (breaking changes) to resolve behavior differences of panels elegantly. 😂🤔

Comment (Leopoldthecoder) [reply packages/date-picker/src/panel/date.vue]: I think picking a date shouldn't change (reset) time.

File: examples/docs/en-US/datetime-picker.md (hunk 4 above)
Comment (RickMacTurk) [reply examples/docs/en-US/datetime-picker.md]: Made above change and resubmitted.

**Comment by wacky6** [comment]
TODO:
* zh_CN, es docs
* update doc attribute's `type` & `accepted value` note

**Comment by Leopoldthecoder** [comment]
I'll do the docs.

**Comment by wuliupo** [comment]
it seems the default time still not work for date picker. (it is for good datetime picker)

**Comment by wacky6** [comment]
it's for datetime not for date.

**Pull request merged**